prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -0,0 +1,313 @@
+                metrics.remove(new MetricInfo(DATE_FORMAT.get().format(date), context.getClusterName()));
+                if (metrics.isEmpty()) {
+                    entityBacklogs.remove(entity);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onFailure(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    @Override
+    public void onStart(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    @Override
+    public void onSuspend(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+    @Override
+    public void onWait(WorkflowExecutionContext context) throws FalconException {
+        // Do Nothing
+    }
+
+
+    /**
+     * Service which executes backlog evaluation and publishing metrics to Graphite parallel for entities.
+     */
+    public static class BacklogMetricEmitter implements Runnable {
+        private ThreadPoolExecutor executor;
+
+        @Override
+        public void run() {
+            executor = new ScheduledThreadPoolExecutor(10);
+            List<Future> futures = new ArrayList<>();
+            try {
+                for (Entity entity : entityBacklogs.keySet()) {
+                    futures.add(executor.submit(new BacklogCalcService(entity, entityBacklogs.get(entity))));
+                }
+                waitForFuturesToComplete(futures);
+            } finally {
+                executor.shutdown();
+            }
+        }
+
+        private void waitForFuturesToComplete(List<Future> futures) {
+            try {
+                for (Future future : futures) {
+                    future.get();
+                }
+            } catch (InterruptedException e) {
+                LOG.error("Interruption while executing tasks " + e);
+            } catch (ExecutionException e) {
+                LOG.error("Error in executing threads " + e);
+            }
+        }
+    }
+
+    /**
+     * Service which calculates backlog for given entity and publish to graphite.
+     */
+    public static class BacklogCalcService implements Runnable {
+
+        private Entity entityObj;
+        private List<MetricInfo> metrics;
+
+        BacklogCalcService(Entity entity, List<MetricInfo> metricInfoList) {
+            this.entityObj = entity;
+            this.metrics = metricInfoList;
+        }
+
+        @Override
+        public void run() {
+            long backlog = 0;
+            MetricInfo metricInfo = null;
+            synchronized (metrics) {
+                long currentTime = System.currentTimeMillis();
+                Iterator iter = metrics.iterator();
+                while (iter.hasNext()) {
+                    try {
+                        metricInfo = (MetricInfo) iter.next();
+                        long time = DATE_FORMAT.get().parse(metricInfo.getNominalTime()).getTime();
+                        backlog += (currentTime - time);
```

Comment:
Backlog calculation has to be per cluster. In this loop, you are not differentiating between clusters. Theoretically, one Falcon Server can manage multiple clusters.